terraform/main.tf

```diff
@@ -163,23 +165,65 @@ resource "google_compute_instance" "logger_vm" {
     }
   }
 
-  # Prometheus config
-  provisioner "file" {
-    source      = "./files/prometheus.yml" # relative to terraform work_dir
-    destination = "./prometheus.yml"       # relative to remote $HOME
-  }
-
   connection {
     type        = "ssh"
     user        = "deployer"
     private_key = file("./deployer_key")
-    host        = self.network_interface.0.access_config.0.nat_ip
+    host        = google_compute_instance.logger_vm.network_interface.0.access_config.0.nat_ip
+  }
+
+  provisioner "file" {
+    source      = "./files/prometheus.yml" # relative to terraform work_dir
+    destination = "./prometheus.yml"       # relative to remote $HOME
   }
 
   provisioner "remote-exec" {
+    connection {
+      type        = "ssh"
+      user        = "deployer"
+      private_key = file("./deployer_key")
+      host        = google_compute_instance.logger_vm.network_interface.0.access_config.0.nat_ip
+    }
+
     scripts = [
       "./scripts/run_minio.sh",
       "./scripts/run_prometheus.sh"
     ]
   }
 }
+
+resource "null_resource" "set_up_docker_tls_and_containerssh" {
```

Comment:
these steps must run after all VMs are created, and need to run in the given order. use `null_resource` could do it